homeassistant/components/azure_devops/sensor.py

```diff
@@ -1,126 +1,97 @@
 
-_LOGGER = logging.getLogger(__name__)
 
-SCAN_INTERVAL = timedelta(seconds=300)
-PARALLEL_UPDATES = 4
+@dataclass
+class AzureDevOpsSensorEntityDescription(
+    AzureDevOpsEntityDescription, SensorEntityDescription
+):
+    """Class describing Azure DevOps sensor entities."""
 
-BUILDS_QUERY = "?queryOrder=queueTimeDescending&maxBuildsPerDefinition=1"
+    attrs: Callable[[DevOpsBuild], Any] = round
+    build_key: int | None = None
+    value: Callable[[DevOpsBuild], StateType] = round
 
 
 async def async_setup_entry(
-    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up Azure DevOps sensor based on a config entry."""
-    instance_key = f"{DOMAIN}_{entry.data[CONF_ORG]}_{entry.data[CONF_PROJECT]}"
-    client = hass.data[instance_key][DATA_AZURE_DEVOPS_CLIENT]
-    organization = entry.data[DATA_ORG]
-    project = entry.data[DATA_PROJECT]
-    sensors = []
-
-    try:
-        builds: list[DevOpsBuild] = await client.get_builds(
-            organization, project, BUILDS_QUERY
-        )
-    except aiohttp.ClientError as exception:
-        _LOGGER.warning(exception)
-        raise PlatformNotReady from exception
-
-    for build in builds:
-        sensors.append(
-            AzureDevOpsLatestBuildSensor(client, organization, project, build)
+    coordinator, project = hass.data[DOMAIN][entry.entry_id]
+
+    sensors = [
+        AzureDevOpsSensor(
+            coordinator,
+            AzureDevOpsSensorEntityDescription(
+                key=f"{build.project.id}_{build.definition.id}_latest_build",
+                name=f"{build.project.name} {build.definition.name} Latest Build",
+                icon="mdi:pipe",
+                attrs=lambda build: {
+                    "definition_id": build.definition.id,
+                    "definition_name": build.definition.name,
+                    "id": build.id,
+                    "reason": build.reason,
+                    "result": build.result,
+                    "source_branch": build.source_branch,
+                    "source_version": build.source_version,
+                    "status": build.status,
+                    "url": build.links.web,
+                    "queue_time": build.queue_time,
+                    "start_time": build.start_time,
+                    "finish_time": build.finish_time,
+                },
+                build_key=key,
+                organization=entry.data[CONF_ORG],
+                project=project,
+                value=lambda build: build.build_number,
+            ),
         )
+        for key, build in enumerate(coordinator.data)
+    ]
 
     async_add_entities(sensors, True)
 
 
-class AzureDevOpsSensor(AzureDevOpsDeviceEntity, SensorEntity):
-    """Defines a Azure DevOps sensor."""
+class AzureDevOpsSensor(SensorEntity, AzureDevOpsDeviceEntity):
+    """Define a Azure DevOps sensor."""
```

Comment:
```suggestion
    """Define a Azure DevOps sensor."""
    
    entity_description: AzureDevOpsSensorEntityDescription
```
This fixes your lint issue and you can then remove the init